src/components/Nav/Nav.js

```diff
@@ -1,7 +1,360 @@
-import React from 'react';
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { useHistory } from 'react-router-dom';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faSearch } from '@fortawesome/free-solid-svg-icons';
+import { faShoppingCart } from '@fortawesome/free-solid-svg-icons';
+import { faUserAlt } from '@fortawesome/free-solid-svg-icons';
 
 function Nav() {
-  return <div></div>;
+  const entry = Object.entries(DATA);
+  const [selectedKey, setSelectedKey] = useState();
+  const [modal, setModal] = useState(false);
+  const history = useHistory();
+  const goToSignUp = () => {
+    history.push('/sign');
+  };
+
+  return (
+    <>
+      <Ad>Ader Error x CASETiFANY 출시 구매하기</Ad>
+      <NavContainter>
+        <a href="/">
+          <Logo src="/images/CASTIFANY.png" />
+        </a>
+        <NavList>
+          {entry.map(el => (
```

Comment:
배열도 구조분해 할당을 할 수 있습니다.
```
const array = ["Hello","World"];
const [first,second] = array;
first // "Hello"
second // "World"
```

```suggestion
          {entry.map([key] => (
```
매개변수자리에서 바로 구조분해할당하면서 이름 지어주면 el + index로 접근하는것보다 가독성이 향상되겠네요!

Reply:
일단 바꾸라는 대로 바꾸긴 했습니다..ㅠㅠ 실행이 되긴 되는데 이해는 안돼요..
구조분해 할당은 이해가 되는데 밑에 key를 집어넣는 부분에서부터 이해가 안됩니다..!

Reply:
콘솔 찍어보고 이해 했습니다!